### PR TITLE
2814 timebox create a poc to test the gmail api

### LINKED ISSUE #2814: [Timebox] Create a POC to test the gmail API
Create a POC to be able to test the email imports from the gmail API.
Verify that the emails are imported correctly in the different tables:
- message data model https://github.com/twentyhq/twenty/issues/2810
- message-thread data model https://github.com/twentyhq/twenty/issues/2811
- message-recipient data model https://github.com/twentyhq/twenty/issues/2812
- message-channel data model https://github.com/twentyhq/twenty/issues/2813
- connected-account data model https://github.com/twentyhq/twenty/issues/2809

### PR DESCRIPTION
**Pull request by bosiraphael**
Ask for user approval to access their email and store the access token and the refresh token in the db

### CODE REVIEW

File: front/src/modules/auth/graphql/mutations/generateShortTermToken.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { gql } from '@apollo/client';
+
+export const GENERATE_ONE_SHORT_TERM_TOKEN = gql`
```
Comment (charlesBochet) [inline front/src/modules/auth/graphql/mutations/generateShortTermToken.ts:3]: Could we change the naming to be Transient Token
It's kind of duplicated with the Login Token but I like to have both separated for now.

So we have:
- TransientToken that can be generated with an accessToken (payload containing workspaceMemberId + workspaceId, we could actually add userId to)
- LoginToken that can be generated by any login method (payload containing email)

File: front/src/modules/settings/data-model/components/SettingsObjectFieldTypeSelectSection.tsx [hunk 2]
```diff
@@ -88,7 +88,7 @@ export const SettingsObjectFieldTypeSelectSection = ({
           FieldMetadataType.Boolean,
           FieldMetadataType.Currency,
           FieldMetadataType.DateTime,
-          FieldMetadataType.Enum,
+          FieldMetadataType.Select,
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/components/SettingsObjectFieldTypeSelectSection.tsx:91]: you are the 3rd person fixing this :p You might have conflict with main, you'll need to rebase or merge main into your branch!

File: front/src/pages/settings/accounts/SettingsAccounts.tsx [hunk 3]
```diff
@@ -1,12 +1,32 @@
...
+
+export const SettingsAccounts = () => {
+  const [generateShortTermToken] = useGenerateShortTermTokenMutation();
+
+  const handleGmailLogin = useCallback(async () => {
+    const authServerUrl = REACT_APP_SERVER_AUTH_URL;
+
+    const shortTermToken = await generateShortTermToken();
+
+    const token =
```
Comment (charlesBochet) [inline front/src/pages/settings/accounts/SettingsAccounts.tsx:18]: transientToken

File: server/.env.example [hunk 4]
```diff
@@ -19,6 +19,11 @@ SIGN_IN_PREFILLED=true
 # REFRESH_TOKEN_EXPIRES_IN=90d 
 # FRONT_AUTH_CALLBACK_URL=http://localhost:3001/verify 
 # AUTH_GOOGLE_ENABLED=false
+# AUTH_GOOGLE_GMAIL_ENABLED=false
```
Comment (charlesBochet) [inline server/.env.example:22]: I think we should change these naming to:
EMAIL_PROVIDER_GMAIL_ENABLED
EMAIL_PROVIDER_GMAIL_CLIENT_ID...

We will also need to add them to the doc

File: server/src/core/auth/dto/save-connected-account.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { ArgsType, Field } from '@nestjs/graphql';
+
+import { IsNotEmpty, IsString } from 'class-validator';
+
+@ArgsType()
+export class SaveConnectedAccountInput {
```
Comment (charlesBochet) [inline server/src/core/auth/dto/save-connected-account.ts:6]: I'm surprised about this class, connectedAccount is a standard object, it should not be in src/core

File: server/src/core/auth/services/google-gmail.service.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    const dataSourceMetadata =
+      await this.dataSourceService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
+        workspaceId,
+      );
+
+    const workspaceDataSource = await this.typeORMService.connectToDataSource(
+      dataSourceMetadata,
+    );
+
+    // Update connected account
```
Comment (charlesBochet) [inline server/src/core/auth/services/google-gmail.service.ts:29]: remove comment :)

**Review by charlesBochet — COMMENTED** [review]
Great progress! I've left comments about naming mainly

File: server/.env.example (hunk 4 above)
Comment (bosiraphael) [reply server/.env.example:22]: Maybe MESSAGING_PROVIDER_... instead, since we will have other messaging systems later, not just emails

**Review by charlesBochet — APPROVED** [review]
LGTM!

**Pull request merged**